package.json

```diff
@@ -55,5 +58,8 @@
     "node": ">=14.0.0"
   },
   "packageManager": "npm@8.10.0",
-  "dependencies": {}
+  "lint-staged": {
+    "*.ts": "eslint --config ./.eslintrc.cjs --fix",
+    "*.{js,jsx,ts,tsx,md,html,css,json}": "prettier --write"
```

Comment:
How does this align with the separate turborepo powered linting that we have in place in scripts already? can the lint staged call the respective npm scripts above?

Also will opening the pattern here beyond `ts, tsx, md` especially to include `js, css` mean that we are running prettier on the contents of `dist/` and other build output folders?

Reply:
This is the same question I ask above, although I'm not sure if my meaning came through as clearly as it does here.

Reply:
The lint-staged package lets us only lint files that were touched during the commit.

I am not sure how to use the existing scripts, since afaik they don't take file arguments.

Reply:
@stevejpurves I tried to make this work, but because the existing formatter grabs *all* files already, I get:

```
> prettier --write "**/*.{ts,tsx,md}" --write /home/stefan/p/myst/myst-theme/packages/site/src/pages/Error404.tsx
```

(where Error404.tsx is the file I touched).

If we systematically go through and add `format` and `format:files`, then I can use the `format:files` target to make it work, and `format` can use `format:files` internally.

Reply:
Re: files included, this is a pre-commit hook, so will only run on files touched in a commit.